varnish/tests/test_varnish.py

```diff
@@ -4,16 +4,39 @@
 
 import pytest
 
+from datadog_checks.dev.utils import get_metadata_metrics
+
 from . import common
 
 pytestmark = [pytest.mark.usefixtures('dd_environment'), pytest.mark.integration]
 
 
+GAUGE_IN_5_RATE_IN_6 = [
+    "varnish.n_expired",
+    "varnish.n_lru_moved",
+    "varnish.n_lru_nuked",
+    "varnish.n_obj_purged",
+    "varnish.n_purges",
+]
+
+
 def test_check(aggregator, check, instance):
     check.check(instance)
-    for mname in common.COMMON_METRICS:
+    exclude = []
+
+    if common.VARNISH_VERSION.startswith("5"):
+        metrics_to_check = common.COMMON_METRICS + common.METRICS_5
+    else:
+        exclude = GAUGE_IN_5_RATE_IN_6
+        metrics_to_check = common.COMMON_METRICS + common.METRICS_6
+
+    for mname in metrics_to_check:
         aggregator.assert_metric(mname, count=1, tags=['cluster:webs', 'varnish_name:default'])
 
+    metadata_metrics = get_metadata_metrics()
+    exclude.extend([m for m in aggregator.metric_names if m not in metadata_metrics])
```

Comment:
https://github.com/DataDog/integrations-core/pull/8049 adds the missing metrics to metadata.csv